llms/mlx_lm/tokenizer_utils.py

```diff
@@ -0,0 +1,66 @@
+import json
+import re
+import shutil
+from pathlib import Path
+
+# Constants copied from transformers library https://github.com/huggingface/transformers/blob/main/src/transformers/tokenization_utils_base.py#L128-L135
+SPECIAL_TOKENS_MAP_FILE = "special_tokens_map.json"
+ADDED_TOKENS_FILE = "added_tokens.json"
+TOKENIZER_CONFIG_FILE = "tokenizer_config.json"
+FULL_TOKENIZER_FILE = "tokenizer.json"
+RE_TOKENIZER_FILE = re.compile(r"tokenizer\.(.*)\.json")
+
+# Hardcoded mapping of model type to vocab file pattern
+VOCAB_FILE_MAPPING = {"qwen": "*.tiktoken"}
```

Comment:
If we want to avoid executing remote code, a hardcoded mapping for the vocabulary may be the best option for now. Otherwise, we may have to dynamically load the remote script and retrieve the name of the vocabulary file from https://huggingface.co/Qwen/Qwen-7B-Chat/blob/main/tokenization_qwen.py#L60.

It will be something similar to https://github.com/huggingface/transformers/blob/main/src/transformers/dynamic_module_utils.py#L380.

@awni What do you think? Personally, I feel that the behavior of the load tokenizer from_pretrained not prompting the user to allow remote code is inconsistent in transformers. I might try raising a PR there and see if they can enable prompting for allowing remote code in the tokenizer as with loading models from_pretrained. Then we would need to hardcode mapping in mlx_lm.